core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -310,33 +310,28 @@ object SparkSubmit extends CommandLineUtils {
       RPackageUtils.checkAndBuildRPackage(args.jars, printStream, args.verbose)
     }
 
-    // In client mode, download remote files.
-    if (deployMode == CLIENT) {
-      val hadoopConf = new HadoopConfiguration()
-      args.primaryResource = Option(args.primaryResource).map(downloadFile(_, hadoopConf)).orNull
-      args.jars = Option(args.jars).map(downloadFileList(_, hadoopConf)).orNull
-      args.pyFiles = Option(args.pyFiles).map(downloadFileList(_, hadoopConf)).orNull
-      args.files = Option(args.files).map(downloadFileList(_, hadoopConf)).orNull
-    }
+    val hadoopConf = new HadoopConfiguration()
+    val targetDir = Files.createTempDirectory("tmp").toFile
+    val sparkConf = new SparkConf(loadDefaults = false).setAll(args.sparkProperties)
```

Comment:
Hmm... I think there might be a problem with logging when calling these classes here. This code is running before the app's main class is called, and it may initialize the logging system (e.g. `Utils.doFetchFile` later calls logging methods in all code paths, initializing the logging system, and so does `SecurityManager`).

This will be a problem, for example, for the spark-shell, which triggers a slightly different code path when initializing logging. Because now the logging system will be initialized by a class that is not the REPL's `Main` class, that will be lost.

Long story short, with the code you're adding, the shell should still be initialized with `WARN` as the default log level. Can you run `spark-shell` with a remote jar in the jars list and verify it still maintains that?